pkg/oc/clusterup/coreinstall/components/pivot/initialconfig.go

```diff
@@ -0,0 +1,144 @@
+package pivot
+
+import (
+	"io/ioutil"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/origin/pkg/oc/clusterup/componentinstall"
+	"github.com/openshift/origin/pkg/oc/clusterup/docker/dockerhelper"
+)
+
+type Component interface {
+	Name() string
+	Install(dockerClient dockerhelper.Interface) error
+}
+
+type KubeAPIServerContent struct {
+	InstallContext componentinstall.Context
+}
+
+func (k *KubeAPIServerContent) Name() string {
+	return "KubeAPIServerContent"
+}
+
+// this ia list of all the secrets and configmaps we need to create
+const (
+	ns = "openshift-kube-apiserver"
+
+	// saTokenSigningCerts contains certificates corresponding to the valid keys that are and were used to sign SA tokens
```

Comment:
To be on the same page:

0. We expect the installer to put the certs into some `/<something>/<certs>` directory.
1. In bootkube phase 1 (underkube), we will call `cluster-kube-apiserver-operator render --cert-dir /<something>/<certs> --bootstrap-assets /<something>/<bootstrap-certs>` to create
  - static pod compatible certs in `/<something>/<bootstrap-certs>` for bootkube phase 1 (underkube)
  - secret + configmap manifests + workload files for the bootkube phase 2 (middlekube)
2. In bootkube phase 2 (middlekube), these secrets and configmaps are created in the cluster and the bootkube daemonsets can pick them up.
3. bootkube succeeds, terminates and give control back to the installer.
4. The installer creates our operators via CVO and our operators adopt these secrets and configmaps.

To make the life of the installer people and us easy, we could have the same directory structure for (0) and mounted into the operator.

Reply:
what are "static pod compatible certs" ?

Reply:
a file structure we can host-mount.